common/authorization/defaultClaimMapper.go

```diff
@@ -0,0 +1,137 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package authorization
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dgrijalva/jwt-go"
+
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/log/loggerimpl"
+	"go.temporal.io/server/common/service/config"
+)
+
+// Default claim mapper that gives system level admin permission to everybody
+type defaultClaimMapper struct {
+	keyProvider TokenKeyProvider
+	logger      log.Logger
+}
+
+func NewDefaultClaimMapper(provider TokenKeyProvider, cfg *config.Config) ClaimMapper {
+	logger := loggerimpl.NewLogger(cfg.Log.NewZapLogger())
+	return &defaultClaimMapper{keyProvider: provider, logger: logger}
+}
+
+var _ ClaimMapper = (*defaultClaimMapper)(nil)
+
+func (a *defaultClaimMapper) GetClaims(authInfo *AuthInfo) (*Claims, error) {
+
+	claims := Claims{}
+
+	if authInfo.authToken != "" {
+		parts := strings.Split(authInfo.authToken, " ")
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("unexpected authorization token format")
+		}
+		if !strings.EqualFold(parts[0], "bearer") {
+			return nil, fmt.Errorf("unexpected name in authorization token: %s", parts[0])
+		}
+		jwtClaims, err := parseJWT(parts[1], a.keyProvider)
+		if err != nil {
+			return nil, err
+		}
+		subject, ok := jwtClaims["sub"].(string)
+		if !ok {
+			return nil, fmt.Errorf("unexpected value type of \"sub\" claim")
+		}
+		claims.subject = subject
+		permissions := jwtClaims["permissions"].([]interface{})
```

Comment:
[Nit] not sure if we want to do a bool check here as well during type assertion

Reply:
Good catch. Fixed and also parameterized the name of permissions claim to make it configurable.